python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py

```diff
@@ -67,17 +68,60 @@ def __init__(
             pipeline_kwargs (dict[str, Any]): Additional keyword arguments passed along
                 to the specific pipeline init (see the documentation for the corresponding pipeline class
                 for possible values). (optional)
+            use_hpu (bool): Whether to use Habana Gaudi HPU for inference. When True, the model will
+                be loaded and run on HPU using GaudiTextGenerationPipeline. (optional)
 
         Note that this model will be downloaded from the Hugging Face model hub.
         """
-        generator = pipeline(
-            task=task,
-            model=ai_model_id,
-            device=device,
-            model_kwargs=model_kwargs,
-            **pipeline_kwargs or {},
-        )
-        resolved_device = f"cuda:{device}" if device >= 0 and torch.cuda.is_available() else "cpu"
+        if use_hpu:
```

Comment:
considering the number of parameters, I think it would be better to allow people to pass in the pipeline, and then they can construct their own GaudiPipeline with the parameters of that, instead of this very hard to discover set of configs

Reply:
that has been something I wanted to do anyway, since that also gives extra capabilities for just HF settings.